hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/ScmTableCountTask.java

```diff
@@ -0,0 +1,212 @@
+import java.util.Map;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+import static org.apache.hadoop.hdds.scm.metadata.SCMDBDefinition.DELETED_BLOCKS;
+import static org.hadoop.ozone.recon.schema.UtilizationSchemaDefinition.SCM_TABLE_COUNT_TABLE_NAME;
+
+
+/**
+ * Any background task that tracks SCM's table counts.
+ */
+public class ScmTableCountTask extends ReconScmTask {
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(ScmTableCountTask.class);
+
+  private final DBStore scmDBStore;
+  private final ScmTableCountDao scmTableCountDao;
+  private final DSLContext dslContext;
+  private final long interval;
+  private ReadWriteLock lock = new ReentrantReadWriteLock(true);
+  private static final long NANOSECONDS_IN_MILLISECOND = 1_000_000;
+
+  public ScmTableCountTask(ReconStorageContainerManagerFacade reconSCM,
+                           ReconTaskStatusDao reconTaskStatusDao,
+                           ReconTaskConfig reconTaskConfig,
+                           ScmTableCountDao scmTableCountDao,
+                           UtilizationSchemaDefinition schemaDefinition) {
+    super(reconTaskStatusDao);
+    this.scmDBStore = reconSCM.getScmDBStore();
+    this.scmTableCountDao = scmTableCountDao;
+    this.dslContext = schemaDefinition.getDSLContext();
+    this.interval = reconTaskConfig.getScmTableCountTaskInterval().toMillis();
+  }
+
+  @Override
+  protected synchronized void run() {
+    try {
+      while (canRun()) {
+        wait(interval);
+        runTableCountProcess();
+      }
+    } catch (Throwable t) {
+      LOG.error("Error while running ScmTableCountTask: {}", t);
+      if (t instanceof InterruptedException) {
+        Thread.currentThread().interrupt();
+      }
+    }
+  }
+
+  private void runTableCountProcess() {
+      try {
+        long startTime, endTime, duration, durationMilliseconds;
+
+        int execute = dslContext.truncate(SCM_TABLE_COUNT_TABLE_NAME).execute();
+        LOG.info("Deleted {} records from {}", execute,
+            SCM_TABLE_COUNT_TABLE_NAME);
+
+        startTime = System.nanoTime();
+        processTableCount();
+        endTime = System.nanoTime();
+        duration = endTime - startTime;
+        durationMilliseconds = duration / NANOSECONDS_IN_MILLISECOND;
+        LOG.info(
+            "Elapsed Time in milliseconds for processTableCount() execution: {}",
+            durationMilliseconds);
+      } catch (Exception e) {
+        LOG.error("An error occurred while performing table count: {}", e);
+      }
+    }
+
+  /**
+   * Processes the table count by iterating over SCM tables and retrieving the
+   * counts of objects in each table. The counts are then stored in the Recon
+   * database.
+   *
+   * @throws IOException if an I/O error occurs during table count processing.
+   */
+  public void processTableCount() throws IOException {
+    // Acquire write lock
+    lock.writeLock().lock();
```

Comment:
Do not see the need of locking here. as this method only called by one thread.

Reply:
Done